components/org.wso2.carbon.identity.auth.service/src/main/java/org/wso2/carbon/identity/auth/service/util/AuthConfigurationUtil.java

```diff
@@ -140,8 +143,13 @@ public void buildClientAuthenticationHandlerControlData() {
             if ( applications != null ) {
                 while ( applications.hasNext() ) {
                     OMElement resource = applications.next();
+                    SecretResolver secretResolver = SecretResolverFactory.create(resource, true);
                     String appName = resource.getAttributeValue(new QName(Constants.APPLICATION_NAME_ATTR));
                     String hash = resource.getAttributeValue(new QName(Constants.APPLICATION_HASH_ATTR));
+                    if (secretResolver.isInitialized()
```

Comment:
Let's add appropriate debug logs. When the secret resolver is not initialized and when the token is / is not protected.

Reply:
> Let's add appropriate debug logs. When the secret resolver is not initialized and when the token is / is not protected.

added debug logs